components/log-viewer-webui/server/src/plugins/MongoReplicaServerPlugin.ts

```diff
@@ -0,0 +1,272 @@
+                    this.collections.set(collectionName, collection);
+                }
+                collection.refAdd();
+
+                socket.data = { collectionName };
+            });
+
+            socket.on("collection::find::toArray", async ({ query, options }, callback) => {
+                const { collectionName } = socket.data;
+                const collection = this.collections.get(collectionName);
+
+                if ("undefined" === typeof collection) {
+                    return callback({
+                        error: "Collection not initialized",
+                    });
+                }
+
+                const documents = await collection.find(query, options).toArray();
+
+                return callback({ data: documents });
+            }
+            );
+
+            socket.on("collection::find::toReactiveArray", async ({ query, options }, callback) => {
+                this.fastify.log.info(`Collection name ${socket.data.collectionName} requested subscription`);
+                const { collectionName } = socket.data;
+                const collection = this.collections.get(collectionName);
+
+                if ("undefined" === typeof collection) {
+                    return callback({
+                        error: "Collection not initialized",
+                    });
+                }
+
+                const { queryHash, watcher } = collection.getWatcher(query, options);
+                callback({ queryHash });
+                watcher.on("change", async () => {
+                    // FIXME: this should be debounced
+                    socket.emit("collection::find::update", {
+                        data: await collection.find(query, options).toArray(),
+                    });
+                });
+
+                socket.emit("collection::find::update", {
+                    data: await collection.find(query, options).toArray(),
+                });
+            }
+            );
+
+            socket.on("collection::find::unsubscribe", ({ queryHash }) => {
+                this.fastify.log.info(`Collection name ${socket.data.collectionName} requested unsubscription`);
+
+                const { collectionName } = socket.data;
+                const collection = this.collections.get(collectionName);
+
+                if ("undefined" === typeof collection) {
+                    return;
+                }
+
+                collection.removeWatcher(queryHash);
+            }
+            );
+        });
+    }
+}
+
+/**
+ * @typedef {object} MongoReplicaServerPluginOptions
+ * @property {string} mongoUri MongoDB URI
+ * @property {string} dbName Database name
+ */
+
+/**
+ * MongoDB replica set plugin for Fastify.
+ *
+ * @param app
+ * @param options
+ * @param options.mongoUri
+ * @param options.dbName
+ */
+// eslint-disable-next-line @stylistic/max-len
+const MongoReplicaServerPlugin = async (app: FastifyInstance, options: {mongoUri: string; dbName: string}): Promise<void> => {
```

Comment:
Return types of `Promise<void>` can be ommited. Since the promise resolves to `void` and the function is `async` (a syntactic sugar for wrapping some return value in a Promise).
```suggestion
const MongoReplicaServerPlugin = async (app: FastifyInstance, options: {mongoUri: string; dbName: string}) => {
```